File: LeetCode_Marathons/LCM5/Hard/Consecutive_Numbers_Sum/a.cpp
class Solution {
public:
    vector<int> getAllDivizors(int n) {
        int sqrtN = sqrt(n);
        vector<int> allDivizors;
        for (int i = 1; i <= sqrtN; i++) {
            if (n % i == 0) {
                allDivizors.push_back(i);
                if (n / i != i) allDivizors.push_back(n / i);
            }
        }
        return allDivizors;
    }

    int consecutiveNumbersSum(int n) {
        vector<int> allDivizors = getAllDivizors(n);
        int counter = 0;
        for (int divizor : allDivizors) {
            if (divizor % 2 == 1) counter++;
        }
        return counter;
    }
};

File: LeetCode_Marathons/LCM5/Hard/Consecutive_Numbers_Sum/a.java
public class Solution {
    private ArrayList<Integer> getAllDivisors(int n) {
        int sqrtN = (int) Math.sqrt(n);
        ArrayList<Integer> allDivisors = new ArrayList<>();
        for (int i = 1; i <= sqrtN; i++) {
            if (n % i == 0) {
                allDivisors.add(i);
                if (n / i != i) {
                    allDivisors.add(n / i);
                }
            }
        }
        return allDivisors;
    }

    public int consecutiveNumbersSum(int n) {
        ArrayList<Integer> allDivisors = getAllDivisors(n);
        int counter = 0;
        for (int divisor : allDivisors) {
            if (divisor % 2 == 1) {
                counter++;
            }
        }
        return counter;
    }
}